docs/current_docs/adopting.mdx

```diff
@@ -4,13 +4,121 @@ slug: /adopting
+
+### Integrating with your dev toolchain
+
+Once your pipeline is daggerized, in the form of one or more Dagger modules, it's time to integrate it with your development toolchain. Dagger is designed for flexibility, so you have many choices in how to integrate it. The downside is that it may feel like *too much* flexibility. What is the *correct* way to integrate?
+
+Here are a few guidelines:
+
+- If your team is willing and able to use the `dagger` CLI directly, we recommend that. Typically that means documenting a few `dagger call` commands in your README, explaining how to use `dagger functions` to discover more functions, and linking to the Dagger documentation to learn more.
+
+- If your team prefers (or is forced to) use another tool, whether off-the-shel like `make`, or custom-built tool or script, it's perfectly fine to wrap your `dagger call` invocations in existing Makefile, configuration files or scripts. Some power users may scratch beneath the surface and call `dagger call` directly; others won't. Both are fine.
+
+- In some rare cases, invoking the `dagger` CLI may not possible or acceptable, and embedding Dagger as a library in an existing codebase might be the best option. This is possible, but should be used as a last resort, as there are limitations. Most notably you will only have access to the core Dagger API, without the ability to extend it with Dagger Modules. We plan to remove this limitation in the future.
+
+### Integrating with CI
+
+Once your pipeline is daggerized, and integrated in your development toolchain, it's time for the final step: integrating it in CI.
+
+Since your daggerized pipelines are packaged as Dagger Modules, and modules only have one dependency - the `dagger` CLI - integrating your daggerized pipelines boils down to three steps:
+
+1. Decide which CI event should trigger which Dagger pipeline
+2. For each pipeline to trigger, determine how to map inputs from the CI environment into arguments to the Dagger functions composing the pipeline
+3. Write the resulting `dagger call` command for each pipeline.
+4. Find the the [integration page](integrations) for your CI platform, and follow the instructions. For example: [GitHub Actions](integrations/github-actions.mdx), [GitLab CI](integrations/gitlab.mdx), [CircleCI](integrations/circleci.mdx), [Jenkins](integrations/jenkins.mdx), [Azure Pipelines](integrations/azure-pipelines.mdx), [AWS CodeBuild](integrations/aws-codebuild.mdx), [Argo Workflows](integrations/argo-workflows.mdx) and [Tekton](integrations/tekton.mdx).
+
+### The Demo
+
+Once your POC is successful, it's time for a victory lap! Make sure to show your team what you accomplished, why, and how. If you'd like help to make your presentation shine, don't hesitate to ask on Discord!
```

Comment:
nit: I think there are too many exclamation points in this section, I'd remove both of the ones here and leave the one at the end of the next paragraph.